src/mo_frontend/typing.ml

```diff
@@ -2048,10 +2049,24 @@ and detect_lost_fields env t = function
          | Some _ -> ()
          | None ->
             warn env fld.at "M0215"
-              "Field `%s` is ignored in record of type%a"
-              id
+              "Field `%s` is ignored in record%s of type%a"
```

Comment:
```suggestion
              "field `%s` is provided but not expected in record%s of type%a"
```
Won't this also warn on regular subtyping? That might be annoying.
One thing you could do is detect if there are similar fields in the record type using the Suggest functions, in case this is a typo.

Reply:
Maybe it's ok, since you only warn on record/object literals, not at every use of subtyping on records.

Reply:
Yeah, it only warns for those where a field is given by the user (either in the record (extension) and it is suppressed bin the typing context, making it unreachable. (Effects survive obviously).

For this PR a `public` in `object`/`class` is given by the user and _immediately shunned by a user-provided type annotation_. This is an indication that the field is either redundant (like above) or if recursively used in the object it could be made `private`.

Reply:
One thing is still missing, namely `let { field } = { field = 4; moot = "MOOT" };` where we don't warn yet. That's because we don't (always) have a pattern type at hand. But we could just reduce to field labels.